src/librustc_mir/interpret/eval_context.rs

```diff
@@ -837,13 +817,39 @@ impl<'a, 'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> EvalContext<'a, 'mir, 'tcx, M
         }
     }
 
+    /// obtains the discriminant from `ByVal` or `ByValPair` and forwards to
+    /// read_discriminant_value for `ByRef`
+    pub fn read_discriminant_value_as_variant_index(
```

Comment:
I'm not sure the comment is correct. Also, `read_discriminant_as_variant_index` seems slightly better as a name?